### PR TITLE
Test Data Generation I -  support a new API to generate skeleton schema/table from query

### PR DESCRIPTION
**Pull request by YannanGao-gs**
#### What type of PR is this?

Improvement
#### What does this PR do / why is it needed ?

Test Data Generation I - support a new API to generate skeleton schema/table from query

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.function.LambdaFunction;
+import org.finos.legend.pure.m3.coreinstance.meta.relational.metamodel.Column;
+import org.finos.legend.pure.m3.coreinstance.meta.relational.metamodel.relation.Table;
+
+import java.util.Collections;
+import java.util.List;
+
+public class TestDataGenerationService
+{
+    public static List<EmbeddedData> generateEmbeddedData(Lambda query, Root_meta_pure_runtime_Runtime runtime, Mapping mapping, PureModel pureModel)
```
Comment (MauricioUyaguari) [inline legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java:38]: can we return a pmcd instead to make it flow better with other generation flows. we can read the embedded data on the studio side.

File: legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/model/TestDataGenerationInput.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+package org.finos.legend.engine.testData.generation.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContext;
+import org.finos.legend.engine.protocol.pure.v1.model.valueSpecification.raw.Lambda;
+
+public class TestDataGenerationInput
+{
+    @JsonProperty
+    public Lambda query;
```
Comment (MauricioUyaguari) [inline legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/model/TestDataGenerationInput.java:26]: lets add a comment that query should not be required but for now we will require it.

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
minor comments.

File: legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java (hunk 1 above)
Comment (YannanGao-gs) [reply legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java:38]: Hello, would you mind being more specific about returning a pmcd?

Comment (MauricioUyaguari) [reply legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java:38]: we should be returning a pmcd in the api.

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**